src/main/java/com/labsynch/labseer/api/ApiSaltController.java

```diff
@@ -82,6 +94,110 @@ public ResponseEntity<String> showJson(@PathVariable("id") Long id) {
 		return new ResponseEntity<String>(salt.toJson(), headers, HttpStatus.OK);
 	}
 
+	@RequestMapping(value = "/edit/{id}", method = RequestMethod.PUT, headers = "Accept=application/json")
+	@ResponseBody
+	public ResponseEntity<String> editSalt(@PathVariable("id") Long id, @RequestBody String json, @RequestParam(value = "dryrun", required = true) boolean dryrun) {
+		Salt oldSalt = Salt.findSalt(id);
+		HttpHeaders headers = new HttpHeaders();
+		headers.add("Content-Type", "application/text; charset=utf-8");
+		headers.add("Access-Control-Allow-Origin", "*");
+		headers.add("Access-Control-Allow-Headers", "Content-Type");
+		headers.add("Cache-Control", "no-store, no-cache, must-revalidate"); // HTTP 1.1
+		headers.add("Pragma", "no-cache"); // HTTP 1.0
+		headers.setExpires(0); // Expire the cache
+		if (oldSalt == null) {
+			return new ResponseEntity<String>("Salt could not be found", headers, HttpStatus.NOT_FOUND);
+		}
+		else
+		{
+			try {
+				Salt newSalt = Salt.fromJsonToSalt(json);
+
+				ArrayList<ErrorMessage> warnings = saltService.validateSaltEdit(oldSalt, newSalt);
+				boolean validSalt = saltService.isValidSaltEdit(oldSalt, newSalt);
+
+				if (validSalt && !dryrun) {
+					try {
+						oldSalt = saltStructureService.edit(oldSalt, newSalt);
+						try
+							{
+								saltService.updateDependencies(oldSalt);
+								// Return Response of Any Warnings / Errors from Validations / Dependencies Done Prior 
+								return new ResponseEntity<String>(ErrorMessage.toJsonArray(warnings), headers, HttpStatus.OK);
+							}
+						catch (Exception e)
+							{
+								return new ResponseEntity<String>("ERROR: Restandardization Issue:" + e.getMessage(), headers,
```

Comment:
nit: this term restandardization is going to throw people off. Can you rename this error to "ERROR: Hit an issue updating salt dependencies:"